### PR TITLE
Added a supervisor to show a spinner and passed time in a long-standing message (CRAFT-100).

### PR DESCRIPTION
**Pull request by facundobatista**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

### CODE REVIEW

File: craft_cli/messages.py [hunk 1]
```diff
@@ -89,6 +99,68 @@ def _get_log_filepath(appname: str) -> pathlib.Path:
     return basedir / filename
 
 
+class _Spinner(threading.Thread):
+    """A supervisor thread that will repeat long-standing messages with a spinner besides it."""
+
+    def __init__(self, printer: "_Printer"):
+        super().__init__()
+        # special flag used to stop itself
```
Comment (cmatsuoka) [inline craft_cli/messages.py:107]: Maybe better as `special flag used to stop the spinner thread`? (otherwise it seems that itself == the flag)

**Review by cmatsuoka — APPROVED** [review]
Interesting logic, not immediately obvious but it seems to work well.

File: craft_cli/messages.py (hunk 1 above)
Comment (facundobatista) [reply craft_cli/messages.py:107]: ah, good idea, thanks!

**Comment by facundobatista** [comment]
Added an explanation on how it works in the class' docstring.

**Review by igorljubuncic — APPROVED** [review]

**Comment by igorljubuncic** [comment]
Looks all right.

**Pull request merged**